### PR TITLE
refactor(renterd): assume autopilot always exists

### PR DESCRIPTION
**Pull request by alexfreska**
- The app was refactored to assume there is a singular autopilot and its API always exists, even if it is disabled.
- Fixed missing fields and structure of bus and autopilot state types.

### CODE REVIEW

**Comment by alexfreska** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/816?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#816** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/816?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/816?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#815** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/815?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#814** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/814?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#810** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#803** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/803?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#763** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/763?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#808** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/808?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#809** <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/809?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by telestrial — APPROVED** [review]

File: apps/renterd/contexts/config/transform.spec.ts [hunk 1]
```diff
@@ -526,29 +518,26 @@ describe('tansforms', () => {
...
-        OS: 'os',
-        buildTime: new Date().getTime(),
-        explorer: {
-          enabled: true,
-          url: 'https://api.siascan.com',
-        },
-        startTime: new Date().getTime(),
+    autopilotState: {
+      id: 'autopilot',
+      configured: true,
```
Comment (peterjan) [inline apps/renterd/contexts/config/transform.spec.ts:523]: This was updated to `enabled`. I imagine this'll be updated again soon and maybe even removed (for the most part).

File: apps/renterd/contexts/config/transform.spec.ts [hunk 2]
```diff
@@ -526,29 +518,26 @@ describe('tansforms', () => {
...
+      scanning: true,
+      scanningLastStart: new Date().toISOString(),
+      synced: true,
+      uptimeMS: '333',
+      network: 'mainnet' as const,
+      version: '0.0.0',
+      commit: 'commit',
+      OS: 'os',
+      buildTime: new Date().getTime(),
+      explorer: {
```
Comment (peterjan) [inline apps/renterd/contexts/config/transform.spec.ts:535]: I was surprised to see this in here. Do you extend our state response with this explorer config?

Comment (alexfreska) [reply apps/renterd/contexts/config/transform.spec.ts:535]: Ah great catch, the bus and autopilot state type definitions look a bit off, I will fix that.

File: apps/renterd/contexts/config/transform.spec.ts (hunk 1 above)
Comment (alexfreska) [reply apps/renterd/contexts/config/transform.spec.ts:523]: This PR adds `enabled` to the autopilot config: https://github.com/SiaFoundation/web/pull/822 

But isn't there still `configured` in terms of autopilot state? I still make use of configured throughout the code to understand if the user has done initial setup. If its gone does this mean there are default values now?

File: libs/renterd-types/src/autopilot.ts [hunk 3]
```diff
@@ -10,13 +10,15 @@ type AutopilotStatus = {
   configured: boolean
   migrating: boolean
   migratingLastStart: string
+  pruning: boolean
```
Comment (alexfreska) [inline libs/renterd-types/src/autopilot.ts:13]: @peterjan fixes to autopilot state

File: libs/renterd-types/src/bus.ts [hunk 4]
```diff
@@ -90,22 +90,22 @@ export const busMultipartPartRoute = '/bus/multipart/part'
...
-  explorer: {
-    enabled: boolean
-    url: string
-  }
 }
 
 export type BusStateParams = void
 export type BusStatePayload = void
 export type BusStateResponse = BuildState & {
+  network: 'mainnet' | 'zen' | 'anagami'
```
Comment (alexfreska) [inline libs/renterd-types/src/bus.ts:103]: @peterjan fixes to bus state

File: apps/renterd/contexts/config/transform.spec.ts (hunk 1 above)
Comment (alexfreska) [reply apps/renterd/contexts/config/transform.spec.ts:523]: Re `configured`: https://github.com/SiaFoundation/renterd/blob/b98cc3d965700f7b675931fb8688e696bfeb1027/api/autopilot.go#L88

**Pull request merged**